crates/ln-dlc-node/src/fee_rate_estimator.rs

```diff
@@ -32,6 +33,43 @@ pub struct FeeRateEstimator {
     fee_rate_cache: RwLock<HashMap<ConfirmationTarget, FeeRate>>,
 }
 
+#[async_trait]
+pub trait EstimateFeeRate {
+    fn estimate(&self, target: ConfirmationTarget) -> FeeRate;
+    async fn update(&self) -> Result<()>;
+}
```

Comment:
🔧 I think this should be defined next to the component that requires this contract i.e. the wallet.

I would keep the `get` method and just call it when implementing the foreign trait method `estimate`.